src/main/java/me/jellysquid/mods/sodium/client/gui/SodiumGameOptions.java

```diff
@@ -89,6 +89,48 @@ public boolean isFancy(GraphicsMode graphicsMode) {
         }
     }
 
+    public enum WeatherQuality implements TextProvider {
+        DEFAULT("options.gamma.default"),
+        FANCY("sodium.options.weather_quality.fancy"),
+        FAST("sodium.options.weather_quality.fast");
+
+        private final Text name;
+
+        WeatherQuality(String name) {
+            this.name = Text.translatable(name);
+        }
+
+        @Override
+        public Text getLocalizedName() {
+            return this.name;
+        }
+
+        public boolean isFancy(GraphicsMode graphicsMode) {
+            return (this == FANCY) || (this == DEFAULT && (graphicsMode == GraphicsMode.FANCY || graphicsMode == GraphicsMode.FABULOUS));
+        }
+    }
+
+    public enum LeavesQuality implements TextProvider {
+        DEFAULT("options.gamma.default"),
+        FANCY("sodium.options.leaves_quality.fancy"),
+        FAST("sodium.options.leaves_quality.fast");
+
+        private final Text name;
+
+        LeavesQuality(String name) {
+            this.name = Text.translatable(name);
+        }
+
+        @Override
+        public Text getLocalizedName() {
+            return this.name;
+        }
+
+        public boolean isFancy(GraphicsMode graphicsMode) {
+            return (this == FANCY) || (this == DEFAULT && (graphicsMode == GraphicsMode.FANCY || graphicsMode == GraphicsMode.FABULOUS));
+        }
+    }
+
```

Comment:
Not fully sure if these enums are still wanted or if I should just do a `Text[]` like it is in clouds.